### PR TITLE
Add ability to configure build service

### PR DESCRIPTION
**Pull request by thiagohmcruz**
Requires:
* https://github.com/jerrymarino/xcbuildkit/pull/49 
* https://github.com/bazel-ios/xchammer/pull/33 

TL;DR:
* Adds `xcbuildkit-data` to `.gitignore` since `xcbuildkit` will use this dir to cache indexing cache
* Load current path to global index store from symbol `GLOBAL_INDEX_STORE_PATH`
* Bump `xcbuildkit`, `xchammer` to pick up changes above and `Tulsi` to pick up lates fixes ([this](https://github.com/bazel-ios/tulsi/pull/7), [this](https://github.com/bazel-ios/tulsi/pull/8))
* `xcodeproj` now accepts `bazel_build_service_config` to configura the build service, for now overriding some paths to allow me to experiment with this for some time before making it fully configurable for customers. Also, adds a test case for this.

- [x] Bump `xcbuildkit` and `xchammer` to main branches before landing

### CODE REVIEW

File: rules/library.bzl [hunk 1]
```diff
@@ -21,6 +21,8 @@ PrivateHeadersInfo = provider(
     },
 )
 
+GLOBAL_INDEX_STORE_PATH = "bazel-out/rules_ios_global_index_store.indexstore"
```
Comment (jerrymarino) [inline rules/library.bzl:24]: This seems reasonable to put here - given we have in-sight dictating the source of truth of these sourcekit invocations form Bazel

Is it far to say that this path needs to be set by Xcode too or are you over-ridding it.

**Review by jerrymarino — APPROVED** [review]

File: rules/library.bzl (hunk 1 above)
Comment (thiagohmcruz) [reply rules/library.bzl:24]: Hep, this is now set in Xcode as well (see `xchammer` PR linked in the PR descr)

**Comment by thiagohmcruz** [comment]
cc @jerrymarino a lot has changed here since your last review. PTAL 🙏

**Comment by jerrymarino** [comment]
Ok let me take a look at it!

File: .gitignore [hunk 2]
```diff
@@ -10,4 +10,5 @@ user.bazelrc
 # XCHammer specific things to ignore for now
 external
 *-XCHammer*.xcodeproj
-xchammer-includes
\ No newline at end of file
+xchammer-includes
+xcbuildkit-data
```
Comment (jerrymarino) [inline .gitignore:14]: The downside of this `xcbuildkit-data` living in the workspace root is everyone has to now add a `.gitignore` for it / see more stuff at the root. `xchammer-includes` was 100% necessary to put there unfortunately

File: tests/ios/app/FW/FW.m [hunk 3]
```diff
@@ -3,4 +3,8 @@
 
 @implementation FW
 
+- (void)foo {
+    NSLog(@"foo");
```
Comment (jerrymarino) [inline tests/ios/app/FW/FW.m:7]: It's probably not a big deal but this will print `foo` . Maybe add something more descriptive like `NSLog(@"Running %s", __PRETTY_FUNCTION__)`.

File: tests/ios/app/App/main.m [hunk 4]
```diff
@@ -2,5 +2,6 @@
 @import OnlySources;
 
 int main(int argc, char **argv) {
+    [[FW alloc] foo];
```
Comment (jerrymarino) [inline tests/ios/app/App/main.m:5]: Minor - in idiomatic objc objects are expected be initialized prior to calling a method on them - instead you can call `+new` or `[[FW alloc] init]`

File: rules/xcodeproj.bzl [hunk 5]
```diff
@@ -2,7 +2,25 @@
...
+
+    We need to better expose the option to
+    customize `GLOBAL_INDEX_STORE_PATH` and expecting to iterate a bit on `indexing_data_dir`
+    and run some tests, once it's battle tested will revert this and allow customers to fully configure
+    """
+    if kwargs_bazel_build_service_config:
+        return bazel_build_service_config(
+            bep_path = kwargs_bazel_build_service_config.bepPath,
+            index_store_path = "$SRCROOT/%s" % GLOBAL_INDEX_STORE_PATH,
+            indexing_data_dir = "$SRCROOT/xcbuildkit-data/indexing",
```
Comment (jerrymarino) [inline rules/xcodeproj.bzl:19]: Any way to we sneak this into an existing known directory or derived data path vs making a new top level dir? I was wondering what the signnificance of the symlink is in the other comment VS just handing it the `GLOBAL_INDEX_STORE_PATH`

**Review by jerrymarino — APPROVED** [review]
Awesome stuff 👏 - a few inline, reviewing the other PRs ATM

File: .gitignore (hunk 2 above)
Comment (thiagohmcruz) [reply .gitignore:14]: We can also put this under `path/to/foo.xcodeproj/xcbuildkit-data`. Thoughts?

File: rules/xcodeproj.bzl (hunk 5 above)
Comment (thiagohmcruz) [reply rules/xcodeproj.bzl:19]: Yeah, moved the DataStore setup to `xcbuildkit` now, see latest commit there. Re `xcbuildkit-data` it's simple to move it to a different place, only added it here since we already had he `xchammer` one. But it's not a requirement for it to live under `$SRCROOT`.

I'll move it to a path under DerivedData, I think that works well.

Comment (thiagohmcruz) [reply rules/xcodeproj.bzl:19]: I might put this under `/tmp` for now

File: tests/ios/app/App/main.m (hunk 4 above)
Comment (thiagohmcruz) [reply tests/ios/app/App/main.m:5]: 👍

File: .gitignore (hunk 2 above)
Comment (thiagohmcruz) [reply .gitignore]: https://github.com/bazel-ios/rules_ios/pull/577#discussion_r1014397768
